michael-chelsea/scripts/article.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+var articles = [];
+
+function Article (rawDataObj) {
+  // DONE: Use the JS object passed in to complete this constructor function:
+  // Save ALL the properties of `rawDataObj` into `this`
+  this.title = rawDataObj.title;
+  this.category = rawDataObj.category;
+  this.author = rawDataObj.author;
+  this.authorUrl = rawDataObj.authorUrl;
+  this.publishedOn = rawDataObj.publishedOn;
+  this.body = rawDataObj.body;
+}
+
+Article.prototype.toHtml = function() {
+  var $newArticle = $('article.template').clone();
+  /* DONE: This cloned article still has a class of template.
+  However, in our modules.css stylesheet, we gave all elements
+  with a class of template a display of none. Let's make
+  sure we're not accidentally hiding our cloned article! */
+  $newArticle.removeClass('template');
+
+  if (!this.publishedOn) $newArticle.addClass('draft');
+  $newArticle.data('category', this.category);
+
+  /* TODO: Now use jQuery traversal and setter methods to fill in the rest
+  of the current template clone with properties from this particular Article instance.
+  We need to fill in:
+    1. author name,---
+    2. author url,---
+    3. article title,---
+    4. article body, and---
+    5. publication date. */
+  // Display the date as a relative number of 'days ago'
+
+  $newArticle.find('h1').html(this.title);
+  $newArticle.find('address').find('a').attr('href', this.authorUrl);
+  $newArticle.find('a').html(this.author);
```

Comment:
- lines 38 & 39 can be combined into one
- instead of .find('address').find('a') you can do .find('address a')